glean-core/src/upload/mod.rs

```diff
@@ -49,6 +47,8 @@ pub enum PingUploadTask {
 pub struct PingUploadManager {
     /// A FIFO queue storing a `PingRequest` for each pending ping.
     queue: Arc<RwLock<VecDeque<PingRequest>>>,
+    /// A manager for the pending pings directories.
+    directory_manager: Arc<RwLock<PingDirectoryManager>>,
```

Comment:
Do we really need it behind a lock and shared?
It's essentially a one-off object solely for processing.
To me it seems like it's better to create it when needed in the thread and keep another instance around in the upload manager

Reply:
> To me it seems like it's better to create it when needed in the thread and keep another instance around in the upload manager.

Hm, I hdan't considered that. I'll give it a try.